### ISSUE TITLE
type_of_target off for labelpowerset and binary relevance strategies

### ISSUE THREAD

**Issue by ChristianSch** [scikit-multilearn/scikit-multilearn#25]
Hi!

While trying to get support for k-fold cross validation (cv) I noticed that the scikit-learn classifiers/strategies use the target classes/shapes as given by `train`. I.e. my `target_true` might look like this: `[array([0, 1, 0, 1, 1, 0, 0, 0])]` and thus the scikit-learn strategies/classifiers return an np-array of targets (in this case 1-D matrix). cv checks for the same target type when trying to score the classifiers.

The problem now is that classifiers of scikit-multilearn return a list of vectors, where `sklearn.utils.multiclass.type_of_target` returns `multiclass-multioutput`, but the sklearn classifiers, as well as my targets are `multilabel-indicator`. Funny thing however is that when you'd print both, they both look like this: `[array([0, 1, 0, 1, 1, 0, 0, 0])]`.

Thus for support of cv in conjunction with scikit-multilearn we'd either need to alter the implementation of `type_of_target` (I might drop an issue on scikit-learn and see whether or not this is expected behaviour or not) or implement casting of predicted targets by transforming the output to np.array.

Any input on this? I'd do a PR on this if I can get a solution to this.

**Comment by ChristianSch** [comment]
As the reply on the scikit-learn issue shows it would be a lot more difficult to alter their implementation of this. I'd push my quickfix if you have not counter arguments. (This is needed for cross validation)

**Comment by niedakh** [comment]
The problem is I've transformed the classes to work with sparse data internally and thus all return a sparse matrix from predict. Could you provide a test that checks for this bug so we know if it also happens on sparse matrices? 

I am having a lot of problems with both working on sparse (enormous gains in speed, label space is usually very sparse), and adhering to scikit API which was designed for a single-label case. I've made a parameter require_dense in classifiers that converts to dense when dealing with a base classifier that does not handle sparse data (basically every single-label classifier in scikit).

I'm wondering, maybe we should do a wrapper that converts sparse to dense for the final outputin crossval? A testcase would help a lot, just a simple one maybe, that runs a CV and catches the error.

**Comment by ChristianSch** [comment]
Ok, working on the test case. There are a few problems I've encountered, see #29 for example. I'm at work right now, but I'll have plenty of time on the weekend.

**Comment by ChristianSch** [comment]
This could take longer, the notebook crashes each time I try to use cross validation (even with try…except). I'm looking into it.

**Comment by ChristianSch** [comment]
Ok, not a problem on our side, for `OneVsRestClassifier` it happens as well.

**Comment by niedakh** [comment]
Fixed in 0.0.3, thank your for your help!
